### PR TITLE
feat: add show/hide password feature for sign-in and login

### PR DESCRIPTION
**Pull request by pratyushsingha**

### CODE REVIEW

File: components/login/LoginForm.tsx [hunk 1]
```diff
@@ -20,6 +20,8 @@ import Link from "next/link";
 import { useRouter } from 'next/navigation';
 import { useState } from "react";
 import { z } from "zod";
+import { AiFillEye, AiFillEyeInvisible } from 'react-icons/ai';
```
Comment (Abbhiishek) [inline components/login/LoginForm.tsx:23]: Hey @pratyushsingha , we use [this ](https://lucide.dev/icons)icon library and wants to maintain this as much as possible.

I hope there is similar icon like this in this library too. do the necessary changes:)

File: components/signup/signupform.tsx [hunk 2]
```diff
@@ -21,8 +21,7 @@ import Link from "next/link";
 import { useRouter } from 'next/navigation';
 import { useEffect, useState } from "react";
 import { z } from "zod";
-
-
+import { AiFillEye, AiFillEyeInvisible } from 'react-icons/ai';
```
Comment (Abbhiishek) [inline components/signup/signupform.tsx:24]: same here too we use lucide icons.

[](https://lucide.dev/icons)

**Review by Abbhiishek — CHANGES_REQUESTED** [review]

**Comment by Abbhiishek** [comment]
check out this : https://lucide.dev/icons/eye

https://lucide.dev/icons/eye-off

**Comment by Abbhiishek** [comment]
image


Also a small fix request to handle a lengthy passwwaord. The eye icon is overlaping with the input fields

File: components/signup/signupform.tsx (hunk 2 above)
Comment (pratyushsingha) [reply components/signup/signupform.tsx:24]: ok

**Comment by pratyushsingha** [comment]
image

**Comment by pratyushsingha** [comment]
"Hi @Abbhiishek, I've addressed the requested changes. Could you please review the pull request again? Thank you!"

File: components/blog/Editor.tsx [hunk 3]
```diff
@@ -28,11 +28,11 @@ const Editor = ({
     title,
     tags,
     content,
-    isdraft,
+    // isdraft,
     setTitle,
     setTags,
     setContent,
-    setIsDraft,
+    // setIsDraft,
```
Comment (Abbhiishek) [inline components/blog/Editor.tsx:35]: revert this back

File: components/blog/Editor.tsx [hunk 4]
```diff
@@ -28,11 +28,11 @@ const Editor = ({
     title,
     tags,
     content,
-    isdraft,
+    // isdraft,
```
Comment (Abbhiishek) [inline components/blog/Editor.tsx:31]: can you provide the reason for the change here

File: components/signup/signupform.tsx [hunk 5]
```diff
@@ -177,18 +178,25 @@ export default function SignUpForm() {
...
-                                type="password"
+                                type={showPassword ? "text" : "password"}
                                 id="password"
                                 placeholder="Password"
                                 value={password}
-                                onChange={(e) => {
-                                    setPasswordStrength(e.target.value.length);
-                                    setPassword(e.target.value);
-                                }}
+                                onChange={(e) => setPassword(e.target.value)}
```
Comment (Abbhiishek) [inline components/signup/signupform.tsx:190]: we have some feature with password strength 

kindly revert back

File: components/ui/label.tsx [hunk 6]
```diff
@@ -7,7 +7,7 @@ import { VariantProps, cva } from "class-variance-authority"
 import { cn } from "@/lib/utils"
 
 const labelVariants = cva(
-  "text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
+  "text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70 mb-3"
```
Comment (Abbhiishek) [inline components/ui/label.tsx:10]: This is a global scope component , changing here can cause a lot of ui bug in different part of site.

you can pass the additional styles as props. Avoid changing the design prototype

File: package.json [hunk 7]
```diff
@@ -77,6 +77,7 @@
     "eslint": "8.40.0",
     "eslint-config-next": "13.4.2",
     "gray-matter": "^4.0.3",
+    "icons": "^1.0.0",
```
Comment (Abbhiishek) [inline package.json:80]: remove the unused dependencies

File: components/login/LoginForm.tsx [hunk 8]
```diff
@@ -134,17 +142,28 @@ export default function LoginForm() {
...
+                        <div className="mb-4 relative">
+                            <Label htmlFor="password" className="mb-2">
+                                Password
+                            </Label>
                             <Input
-                                type="password"
+                                type={showPassword ? "text" : "password"}
                                 id="password"
                                 placeholder="Password"
                                 value={password}
```
Comment (Abbhiishek) [inline components/login/LoginForm.tsx:156]: you can pass class (for styles here ), that would be applied just for this specific input field

File: yarn.lock [hunk 9]
```diff
@@ -5522,6 +5522,11 @@ husky@^8.0.3:
   resolved "https://registry.npmjs.org/husky/-/husky-8.0.3.tgz"
   integrity sha512-+dQSyqPh4x1hlO1swXBiNb2HzTDN1I2IGLQx1GrBuiqFJfoMrnZWwVmatvSiO+Iz8fBUnf+lekwNo4c2LlXItg==
 
+icons@^1.0.0:
```
Comment (Abbhiishek) [inline yarn.lock:5525]: remove unused dependencies

**Review by Abbhiishek — CHANGES_REQUESTED** [review]
Kindly make the mentioned changes. Rest looks good to me.

Also push your details in this file with proper details.
The file where you have to add your details : 
`config/contributors.ts`

Make sure you follow the type defination as follow : 
```ts
interface IContributors {
    name: string;
    role: string;
    issueLink: string;
    prLink: string;
    githubAccountUrl: string;
    rewviewWords: string;
}
```

Happy Hacking 🙌

File: components/ui/label.tsx (hunk 6 above)
Comment (Abbhiishek) [reply components/ui/label.tsx:10]: also kindly note that styles works but the text is wrapped and last 4 digits are not shown cause it's overlaped with the icon 

image

Have a look in the poicture you can too generate the issue

type : `1111111111111111111111111111111111111111111111111111111111222`

The last 2 digits are not visible. Kindly manage this scope

**Review by Abbhiishek — CHANGES_REQUESTED** [review]

File: components/signup/signupform.tsx [hunk 10]
```diff
@@ -177,18 +178,26 @@ export default function SignUpForm() {
...
                                 id="password"
-                                placeholder="Password"
+                                placeholder="password"
                                 value={password}
-                                onChange={(e) => {
-                                    setPasswordStrength(e.target.value.length);
-                                    setPassword(e.target.value);
-                                }}
+                                onChange={(e) => setPassword(e.target.value)}
+                                className="pl-3 pr-10 py-2 rounded-lg"
```
Comment (Abbhiishek) [inline components/signup/signupform.tsx:190]: All this class name are not necessary we already have those defined in global component/ui.

Kindly remove unwanted class names

**Review by Abbhiishek — CHANGES_REQUESTED** [review]

**Comment by pratyushsingha** [comment]
this PR was accidently closed my me . created a new branch login-signup and here is the new [PR #27 ](https://github.com/Abbhiishek/codecommunitymusic-client/pull/27)

**Pull request closed without merging**